Doc feedback and suggestions

chreekat opened the pull request:
I have sprinkled a lot of FIXMES throughout that I could use some assistance with!

* [README rendered](https://github.com/chreekat/error-message-index/blob/wip/b/doc-feedback/README.md)
* [CONTRIBUTING rendered](https://github.com/chreekat/error-message-index/blob/wip/b/doc-feedback/CONTRIBUTING.md)

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -7,63 +7,73 @@ are in force.
+title: [FIXME: Is this something written by the contributor, or does it come from the tool?]
+summary: [FIXME: Is this something written by the contributor, or does it come from the tool?]
+severity: [FIXME]: What are the possible values here?
+introduced: <!-- Optional, but good to add if you know when the message was added to the tool -->
```
This is when the code was added, not the message!

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -7,63 +7,73 @@ are in force.
+`message-index/messages/GHC-00000/index.md`, and write your documentation there.
 
+Here is a template you can use for the file:
+[FIXME: providing a tool would be a lot nicer]
```
I'm not quite sure what is meant here. Can you elaborate?

chreekat replied:
I meant I would like to replace "Here is a template you can use..." with "Here are the instructions for initializing new file." I guess if #408 takes flight, it could be used here.

david-christiansen commented on CONTRIBUTING.md:
````diff
@@ -7,63 +7,73 @@ are in force.
-## How to Document a GHC Error Code
+```
+---
+title: [FIXME: Is this something written by the contributor, or does it come from the tool?]
````
This is written by the contributor

david-christiansen replied:
It's used as the title for the docs

david-christiansen commented on CONTRIBUTING.md:
````diff
@@ -7,63 +7,73 @@ are in force.
+```
+---
+title: [FIXME: Is this something written by the contributor, or does it come from the tool?]
+summary: [FIXME: Is this something written by the contributor, or does it come from the tool?]
````
Also written by the contributor - it's used in the "summary" column in the table on the main page

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -7,63 +7,73 @@ are in force.
+---
+title: [FIXME: Is this something written by the contributor, or does it come from the tool?]
+summary: [FIXME: Is this something written by the contributor, or does it come from the tool?]
+severity: [FIXME]: What are the possible values here?
```
`error` and `warning`

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -7,63 +7,73 @@ are in force.
 
 The Haskell Message Index is generated from a collection of files on
-disk using Hakyll. Inside the top-level of the site source, there is a
+disk using Hakyll. From these files, descriptions of the message are generated,
+as well as examples of how to handle them.
```
```suggestion
disk using Hakyll. The contents of these files, including the overall description and the examples,
is used to generate both the individual description pages and the main page of the site.
```

david-christiansen replied:
(the examples aren't generated from the files - the files contain the examples)

chreekat replied:
Oh—as I see it, the examples *are* generated from the files, because the example is the whole shebang

![image](https://user-images.githubusercontent.com/538538/233352580-a6663f38-909a-4ef5-8ddd-e1b01141e0c4.png)

Taken as a whole, that box isn't contained in any one file.

david-christiansen replied:
I think that this kind of hair-splitting could end up going either way, so we should find a formulation that isn't as easily mis-parsed. What about "The complete documentation page is generated from the written description and the example files." or something like that?

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -7,63 +7,73 @@ are in force.
+disk using Hakyll. From these files, descriptions of the message are generated,
+as well as examples of how to handle them.
+
+Inside the top-level of the site source, there is a
```
```suggestion
Inside the top level of the site source, there is a
```

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -90,6 +100,8 @@ named as specified here, while the other components may vary.
 
 You can also use the `makeFolder.sh` templating script. Usage is as follows:
 
+[FIXME: makeFolder.sh is GHC-specific]
```
I think #408 can take care of this nicely.

chreekat replied:
Excellent :)

david-christiansen commented on CONTRIBUTING.md:
```diff
@@ -137,9 +149,24 @@ field. All `.hs` files are shown in the list of files for the
-Pull requests that make it easier to understand or navigate are very
-welcome. The main generator `site.hs` is formatted using
-[Ormolu](https://github.com/tweag/ormolu).
+[FIXME I don't know how, or why, to follow these steps. Can someone who's submitted a
+message explain it?]
```
As far as "how" goes, it relies on having the GHC source available. I think that we used to have people get that in this guide (which was written before the error codes branch was merged, which was before the 9.6 release), but now we need to clean it up.

The "why" is that it's not always clear how to trigger a message simply based on its string rendering. The GHC test suite, however, is a good source of programs that exhibit the message. Grepping the "expected errors" text is a way to find examples.

We should probably go through this and see if there's a better way since Sam's reworking of the code assignment mechanism. Probably grepping the tests for the code is enough now.

chreekat replied:
I can't remember Sam's handle, but do you think he might want to weigh in? Should this section just be removed, perhaps?

david-christiansen replied:
I don't know what he's called over here - he's sheaf on Gitlab. But I don't think we need his feedback - we can just try grepping the test suite for messages that include the desired code string :-)

david-christiansen commented on README.md:
```diff
@@ -1,7 +1,23 @@
+
+## Supported Tools
+
+Today, the Haskell Message Index supports three tools, with more (like [Cabal](https://github.com/haskell/cabal/issues/8543)) on the horizon.
```
```suggestion
Today, the Haskell Message Index supports three tools. Any user-facing Haskell-related programming tools are welcome to join the effort!
```

I don't want to make promises on behalf of other projects. I think Cabal is positive about the idea, but they don't presently have someone implementing it.

david-christiansen reviewed (changes requested):
Thanks! This document really does need a reworking now that it's not really intended for a room at Zurihac '22 anymore. And fresh eyes are the best eyes!

chreekat reviewed (commented):
Thanks! I can work in some of these suggestions.

chreekat commented:
I've made some corrections. There are still two fixmes. Since #408 would make life easier for newcomers, I am considering waiting until it's in and then updating this one. But that's probably an excess of perfectionism and too much of a burden, so maybe I'll just remove the FIXME and call it good for now.

BinderDavid commented:
#408 is now merged, so I would like to help push this PR over the finish line :)

One additional thing I noticed is that the information on how to improve/contribute to the site is distributed at three different places:
- Readme.md contains the information on how to build and run the site locally
- tech-choices.md contains the information about what technology is used for the site.
- CONTRIBUTING.md contains the section "Contributing to the Site"

I suggest we merge this information, preferably within `Contributing.md`

chreekat commented:
@BinderDavid I'm done adding references to create-message-template. I also moved the instructions on running and building the site into CONTRIBUTING.

The only other file that *may* be moved into CONTRIBUTING is  `tool-developers.md`, but I don't think it's critical. It's distinct enough as a topic that it's probably fine in a separate document, anyway.

I tweaked the output from `create-message-template` a little in [fb7264a](https://github.com/haskellfoundation/error-message-index/pull/409/commits/fb7264a7bd7fff8cbe76fd45e5f6f0bac5c66f6f) and added a single golden test in [e84b10e](https://github.com/haskellfoundation/error-message-index/pull/409/commits/e84b10e414215a832b7e7688f31ab080a076729b) -- hope you don't mind. :)

From my perspective this is ready to merge now.

BinderDavid commented:
> @BinderDavid I'm done adding references to create-message-template. I also moved the instructions on running and building the site into CONTRIBUTING.
> 
> The only other file that _may_ be moved into CONTRIBUTING is `tool-developers.md`, but I don't think it's critical. It's distinct enough as a topic that it's probably fine in a separate document, anyway.
> 
> I tweaked the output from `create-message-template` a little in [fb7264a](https://github.com/haskellfoundation/error-message-index/pull/409/commits/fb7264a7bd7fff8cbe76fd45e5f6f0bac5c66f6f) and added a single golden test in [e84b10e](https://github.com/haskellfoundation/error-message-index/pull/409/commits/e84b10e414215a832b7e7688f31ab080a076729b) -- hope you don't mind. :)
> 
> From my perspective this is ready to merge now.

Thanks! I'll be looking at this tonight. I'm currently not at home :)

BinderDavid reviewed (approved):
Thanks. That looks all very good :) The test for the scaffolding tool is not yet enabled in CI, but I will just open a separate PR for this.

chreekat commented:
Thanks!!
